src/modules/producer.js

```diff
@@ -165,14 +165,27 @@ class Producer {
     // messages are persistent
     options.persistent = true;
 
+    // convert timeout to amqp's expiration. It's message-level expiration.
+    // The message will be discarded from a queue once it’s been there longer than the given number of milliseconds
+    // This is needed to avoid the case when the message which is already expired from our pov (via timeout)
+    // is still waiting in the queue and thus is about to be processed by the consumer.
+    // Unfortunately, we can do nothing if the message is already consumed and is being processed at the moment
+    // when the timeout appears.
+    if (options.timeout && options.timeout > 0) {
```

Comment:
Don't understand why we need this new `timeout` option while you can use existing working `expiration`.

Reply:
The `timeout` option here is used to as an override to the `rpcTimeout` connection configuration, which creates the "promise race" on the `prepareTimeoutRpc` method. It does need to set the `expiration` amqplib setting as well when used.

As the readme says:
> The optional `timeout` option results in a rejection when no answer has been received after the given amount of milliseconds.
> When '0' is given, there will be no timeout for this call.
> This value will overwrite the default timeout set in the config in `rpcTimeout`.


I agree it would be preferable to use `expiration` directly instead, in order to better match amqplib, but we can't just remove the `timeout` key and break the current api.
Maybe we should deprecate the `timeout` option and the `rpcTimeout` configuration and switch to `expiration` and `rpcExpiration` to make it clearer?

What do you think @yosiat @sergejivanoff6 ?

Reply:
@yosiat , I agree with Ram that we cannot completely remove timeout if we want to be compatible with the current version of the lib. Talking about the deprecation of timeout - I like this idea.

Reply:
In my opinion, we changing a behaviour here. I think adding a new option `expiration` in addition to `timeout`, makes more sense. This way people can migrate to new behaviour as they see fit.

That's said, I do think we should add a deprecation message to `timeout` option and write a warning that `timeout` is deprecated and `expiration` should be used instead. At some point we will tag a major version that remove `timeout` option from the library.

@ramhr @yosiat wdyt?

Reply:
@ramhr @yosiat wdyt?

Reply:
@shamil I love your option, @sergejivanoff6 what do you think? can you adapt to shamil's proposed solution?

Reply:
Hey @yosiat 
I consider this PR as not like adding new functionality but as a bug fix. I think message-level timeout was a mistake not only because we have amqp standard `expiration` exactly for this purpose but also because it can be confused with the connection-level timeout which is about waiting before reconnecting to rabbitMQ. The reason why we cannot just replace it completely in this version, as @ramhr said, is that we have to be compatible with the existing clients. I agree that it makes sense to remove `timeout` in the major release. 
I've added a deprecation message for using `timeout` in log and made a corresponding note in README.
I don’t think though that `expiration` and `timeout` should be independent of each other as in fact they're about the same. It's hard to imagine someone who explicitly wants to set a timeout that rejects the promise on the producer side but doesn't affect the consumer at all. So with this PR being merged, you can use `expiration` and `timeout` for rpc calls and result will be the same.
I've aligned to what @shamil and @ramhr said, I guess. Let me know if you don't agree. Thanks